src/github.com/matrix-org/dendrite/clientapi/routing/createroom.go

```diff
@@ -43,10 +42,30 @@ type createRoomRequest struct {
 	Topic           string                 `json:"topic"`
 	Preset          string                 `json:"preset"`
 	CreationContent map[string]interface{} `json:"creation_content"`
-	InitialState    json.RawMessage        `json:"initial_state"` // TODO
+	InitialState    []fledglingEvent       `json:"initial_state"`
 	RoomAliasName   string                 `json:"room_alias_name"`
+	GuestCanJoin    bool                   `json:"guest_can_join"`
 }
 
+const (
+	presetPrivateChat        = "private_chat"
+	presetTrustedPrivateChat = "trusted_private_chat"
+	presetPublicChat         = "public_chat"
+)
+
+const (
+	joinRulePublic = "public"
+	// joinRuleCanJoin = "can_join"
+	joinRuleInvite = "invite"
+	// joinRuleKnock   = "knock"
+	// joinRulePrivate = "private"
```

Comment:
I believe `private` is implemented so we can uncomment it, and I wouldn't bother with keeping knock or can_join there so just delete them.

Reply:
`private` isn't used in this file, and I haven't found any other occurrence across other files.
Also [the doc](https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#m-room-join-rules) states that `private` is an unimplemented keyword (and I found no implementation in Synapse)

I think there should be only `public` and `invite` here.

Btw `can_join` was just a terrible mix up/copy paste error (it's a `guest_access` value)

Reply:
Oh, hmm, empirically `private` does work on synapse I think, but if its not specced then yeah, may as well leave it out for now.